cupy/core/core.pyx

```diff
@@ -70,6 +70,29 @@ cdef inline _should_use_rop(x, y):
 
 
 cdef tuple _HANDLED_TYPES
+cdef int fp32_compute_type = FP32_COMPUTE_TBD
+
+
+cpdef set_fp32_compute_type(type):
+    global fp32_compute_type
+    cdef int compute_capability = int(device.get_compute_capability())
+    if type in (FP32_COMPUTE_TBD, FP32_COMPUTE_DEFAULT, FP32_COMPUTE_PEDANTIC):
+        fp32_compute_type = type
+    elif type == FP32_COMPUTE_FAST_TF32:
+        if compute_capability >= 80:
+            fp32_compute_type = type
+        else:
+            fp32_compute_type = FP32_COMPUTE_DEFAULT
+            warnings.warn('FP32_COMPUTE_FAST_TF32 is only available on GPUs'
+                          ' with compute capability 8.0 or higher.'
+                          ' FP32_COMPUTE_DEFAULT will be used instead.')
+    else:
+        raise ValueError('Unknown fp32 compute type: {}'.format(type))
+
+
+cpdef get_fp32_compute_type():
+    global fp32_compute_type
+    return fp32_compute_type
```

Comment:
I think it's better to do this here:

```py
if fp32_compute_type == FP32_COMPUTE_TBD:
   if int(os.getenv('CUPY_TF32', '0')) > 0:
       set_fp32_compute_type(FP32_COMPUTE_FAST_TF32)
  else:
      set_fp32_compute_type(FP32_COMPUTE_DEFAULT) 
```

Reply:
Agreed. I will move the initialization process there.